apps/system/js/task_manager.js

```diff
@@ -732,6 +721,12 @@
    * @param {DOMNode} element
    */
   TaskManager.prototype.getCardForElement = function(element) {
+    while (element && element !== this.element) {
+      if (element.classList.contains('card')) {
+        break;
+      }
+      element = element.parentNode;
+    }
```

Comment:
In _updateCardPositions, as we pass in the card element so its basically just a look up in his.cardsByAppID. But we use the same method in handleTap where we get a descendent node, so I think we need it.